components/coding/profileV2/achievement_components/AchievementComponent.tsx

```diff
@@ -0,0 +1,120 @@
+import React, { useState } from "react";
+import { PencilAltIcon } from "@heroicons/react/outline";
+import { useMutation, useQuery } from "@apollo/client";
+import {
+  FetchBadgeResponse,
+  FETCH_CODING_BADGES,
+  IntroCourseUnit,
+} from "../../../../graphql/coding/userBadges/fetchUserBadges";
+import { Button } from "../../../ui/Button";
+import {
+  INSERT_USER_CODING_BADGES,
+  DELETE_USER_CODING_BADGES,
+} from "../../../../graphql/coding/userBadges/updateUserCodingBadges";
+import UnitBadgeSection from "./UnitBadgeSection";
+import findBadgeDiff from "./findBadgeDiff";
+
+export type AchievementComponentProps = {
+  userId: string;
+  isEditable: boolean;
+};
+
+const AchievementComponent = ({
+  userId,
+  isEditable,
+}: AchievementComponentProps) => {
+  const [unitBadges, setUnitBadges] = useState<IntroCourseUnit[]>();
+  const [editMode, setEditMode] = useState(false);
+  const { data } = useQuery<FetchBadgeResponse>(FETCH_CODING_BADGES, {
+    variables: {
+      userId: userId,
+    },
+    onCompleted: (data) => {
+      setUnitBadges(data.intro_course_unit);
+    },
+  });
+
+  const [saveAddedBadges] = useMutation(INSERT_USER_CODING_BADGES, {
+    refetchQueries: [{ query: FETCH_CODING_BADGES }],
+  });
+  const [saveRemovedBadges] = useMutation(DELETE_USER_CODING_BADGES, {
+    refetchQueries: [{ query: FETCH_CODING_BADGES }],
+  });
+  // move handler inside AC component, and add type props
+  // keep findBadgeDiff
+  const handleOnSaveButtonClick = () => {
+    const addedBadges = findBadgeDiff(
+      data.intro_course_unit,
+      unitBadges,
+      userId
+    ).addedBadges;
+    const removedBadges = findBadgeDiff(
+      data.intro_course_unit,
+      unitBadges,
+      userId
+    ).removedBadges;
+
+    if (addedBadges.length > 0) {
+      saveAddedBadges({
+        variables: {
+          objects: addedBadges,
+        },
+      });
+    }
+    if (removedBadges.length > 0) {
```

Comment:
this is obviously not the most ideal since we can't bulk delete a list of objects, but i love that you came up with this workaround for deleting... might be the next best thing we can do!